src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -7951,6 +7937,10 @@ void StorageReplicatedMergeTree::createTableSharedID() const
             id = zookeeper->get(zookeeper_table_id_path);
             LOG_DEBUG(log, "Shared ID on path {} concurrently created, will set ID {}", zookeeper_table_id_path, id);
         }
+        else if (code == Coordination::Error::ZNONODE) /// table completely dropped, we can choose any id we want
+        {
+            LOG_DEBUG(log, "Table was completely drop, we can use anything as ID (will use {})", id);
```

Comment:
Why do we need an id if the table is completely dropped? What are we going to unlock in `unlockSharedDataByID` in this case?

Reply:
We will just check that this ID doesn't exist in ZK and assume parts unlocked, but I'm not 100% sure...